bots/run-queue

```diff
@@ -56,41 +109,37 @@ def main():
     channel = connection.channel()
 
     try:
+        declare_webhook_result = channel.queue_declare(queue='webhook', passive=True, auto_delete=False)
         declare_public_result = channel.queue_declare(queue='public', passive=True, auto_delete=False)
         declare_rhel_result = channel.queue_declare(queue='rhel', passive=True, auto_delete=False)
     except pika.exceptions.ChannelClosed:
-        print("One of the queues doesn't exist")
         return 1
 
-    queue='public'
-    if redhat_network():
-        # Try the rhel queue if the public queue is empty
-        if declare_public_result.method.message_count == 0:
-            queue = 'rhel'
-        # If both are non-empty, shuffle
-        elif declare_rhel_result.method.message_count > 0:
-            queue = ['public', 'rhel'][random.randrange(2)]
+    if not any([declare_webhook_result.method.message_count,
+                declare_public_result.method.message_count,
+                declare_rhel_result.method.message_count]):
+        logging.warn("All queues are empty")
+        return 1
 
-    # Get one item from the queue if present
-    method_frame, header_frame, body = channel.basic_get(queue=queue)
-    if method_frame:
-        ret = 0
-        body = json.loads(body)
-        sys.stderr.write("Consuming {0} task:\n{1}\n".format(body['type'],json.dumps(body, indent=2, sort_keys=True)))
-        sys.stderr.flush()
-        p = subprocess.Popen(body['command'], shell=True)
-        while p.poll() is None:
-            connection.process_data_events()
-            time.sleep(3)
-
-        if p.returncode in [0, 2]:
-            channel.basic_ack(method_frame.delivery_tag)
+    if random.randrange(2):
+        cmd, body = consume_webhook_queue(channel, amqp)
     else:
-        # nothing in queue, no work done
-        ret = 1
+        cmd, body = consume_task_queue(channel, amqp, declare_public_result, declare_rhel_result)
+
+    if not cmd:
+        logging.warn("Unable to construct command")
+        return 1
+
+    logging.info("Consuming {0} task:\n{1}\n".format(body['type'], json.dumps(body, indent=2, sort_keys=True)))
```

Comment:
Let's log the generated cmd. Dumping the entire body OTOH is really noisy -- is that just an intermediate debugging thing? I'd leave it out when it lands in production.

Reply:
we did this before since the body was small, but now it contains the entire request, so you're right; i'll drop this.